ci.ps1

```diff
@@ -0,0 +1,149 @@
+    Script         = { bundle exec r10k puppetfile install }
+  }
+
+  Invoke-ScriptBlockWithRetry @retryArgs
+}
+
+function New-RandomPassword
+{
+  Add-Type -AssemblyName System.Web
+  "&aA4" + [System.Web.Security.Membership]::GeneratePassword(10, 3)
+}
+
+function New-LocalAdmin($userName, $password)
+{
+  $userArgs = @{
+    Name     = $userName
+    Password = (ConvertTo-SecureString -String $password -Force -AsPlainText)
+  }
+
+  $user = New-LocalUser @userArgs
+  Write-Information ($user | Format-List | Out-String)
+  Add-LocalGroupMember -Group 'Remote Management Users' -Member $user
+  Add-LocalGroupMember -Group Administrators -Member $user
+}
+
+function Install-Certificate($path, $password)
+{
+  $importArgs = @{
+    FilePath          = $path
+    CertStoreLocation = 'cert:\\LocalMachine\\My'
+    Password          = (ConvertTo-SecureString -String $password -Force -AsPlainText)
+  }
+
+  return (Import-PfxCertificate @importArgs)
+}
+
+function Grant-WinRMHttpsAccess($certThumbprint)
+{
+  $winRMArgs = @{
+    ResourceURI = 'winrm/config/Listener'
+    SelectorSet = @{ Address = '*'; Transport = 'HTTPS' }
+    ValueSet    = @{ Hostname = 'localhost'; CertificateThumbprint = $certThumbprint }
+  }
+  $instance = New-WSManInstance @winRMArgs
+  Write-Information ($instance | Format-List | Out-String)
+}
+
+function Set-WinRMHostConfiguration
+{
+  # configure WinRM to use resources/cert.pfx for SSL
+  $cert = Install-Certificate -Path 'resources/cert.pfx' -Password 'bolt'
+  Write-Information ($cert | Format-List | Out-String)
+  Grant-WinRMHttpsAccess -CertThumbprint $cert.Thumbprint
+}
+
+function Invoke-ScriptBlockWithRetry([ScriptBlock]$script, $failMessage, $successMessage, $retries = 15, $timeout = 1)
+{
+  $retried = 0
+
+  Do
+  {
+    try {
+      $script.Invoke()
+      Write-Information "$successMessage after $($retried + 1) attempt(s)"
+      return $true
+    }
+    catch
+    {
+      $retried++
+      Start-Sleep -Seconds $timeout
+    }
+  } While ($retried -lt $retries)
+
+  throw "ERROR: $failMessage in $retried retries`n$($Error[0])"
+
+}
+
+function Test-WinRMConfiguration($userName, $password, $retries = 15, $timeout = 1)
+{
+  $retryArgs = @{
+    FailMessage    = 'Failed to establish WinRM connection over SSL'
+    SuccessMessage = "Successfully established WinRM connection with $userName"
+    Retries        = $retries
+    Timeout        = $timeout
+    Script         = {
+      $pass = ConvertTo-SecureString $password -AsPlainText -Force
+      $sessionArgs = @{
+        ComputerName = 'localhost'
+        Credential   = New-Object System.Management.Automation.PSCredential ($userName, $pass)
+        UseSSL       = $true
+        SessionOption = New-PSSessionOption -SkipRevocationCheck -SkipCACheck
+      }
+
+      if (New-PSSession @sessionArgs) { return $true }
+    }
+  }
+
+  Invoke-ScriptBlockWithRetry @retryArgs
+}
+
+# Ensure Puppet Ruby 5 / 6 takes precedence over system Ruby
```

Comment:
IIRC these paths were to ensure that puppet ruby was picked up for the `basic::ruby_task`. Now that we are able to configure `implementations` for winrm transport we should be able to handle this in 
```ruby
puppet_5_config = config
puppet_5_config['winrm']['implementations'] = { '.rb' =>  "#{ENV['ProgramFiles'}\Puppet Labs\Puppet\sys\ruby\bin" }
with_tempfile_containing('bolt', YAML.dump(puppet_5_config), '.yaml') do |conf|
```
in https://github.com/puppetlabs/bolt/blob/b062fa3faed7fa8600ef17275a46933b37a12bc6/spec/integration/apply_spec.rb#L278-L279

Reply:
I'd say let's leave that for a follow up because this PR only touches `appveyor.yml`, which makes it simpler to vet / merge.

Reply:
That sounds good. Once we see this merged and green I can circle back on that.